app/models/solidus_content/entry_type.rb

```diff
@@ -19,4 +26,32 @@ def content_for(entry)
   def provider
     SolidusContent.config.providers[provider_name.to_sym]
   end
+
+  def provider_name_readonly?
+    persisted?
+  end
+
+  def options=value
+    if String === value
+      begin
+        value = JSON.parse(value)
+      rescue JSON::ParserError
+        value
```

Comment:
Should we swallow errors like this? What's the use case for non-JSON content in the `options` attribute?

Reply:
You're right, there is an underlying problem that is: how to accept with serialized  json? where should we handle it before feeding it to the model?

I'm now updating the code to use the ActiveRecord Json type to do the serialization, [it still has a `rescue nil`](https://github.com/rails/rails/blob/6-0-stable/activerecord/lib/active_record/type/json.rb#L14) in it, but it's library code and no one cares about that 😄, jk, library code is the place in which to do this kind of stuff 👍